spidermon/contrib/scrapy/monitors.py

```diff
@@ -44,6 +44,38 @@ def test_minimum_number_of_items(self):
         self.assertTrue(item_extracted >= self.minimum_threshold, msg=msg)
 
 
+@monitors.name("Extracted Expected Items Monitor")
+class ExpectedItemCountMonitor(BaseScrapyMonitor):
+    """Check if spider extracted the expected minimum number of items provided
+    in a stats `spidermon/expected_item_scraped_count`.
+
+    You can configure it using ``SPIDERMON_EXPECTED_ITEMS_THRESHOLD`` setting.
+    The default value is 0, meaning that the number of scraped items should be
+    equal to the expected one.
+    Any other value in the range [0, 1] will be computed as a percentage
+    threshold and subtracted from `spidermon/expected_item_scraped_count`.
+    """
+
+    def run(self, result):
+        settings = self.crawler.settings
+        self.expected_items_threshold = settings.getfloat('SPIDERMON_EXPECTED_ITEMS_THRESHOLD', 0)
+        if not (0.0 <= self.expected_items_threshold <= 1.0):
+            raise NotConfigured("You must specify a threshold of expected items.")
```

Comment:
For me both ItemCount monitors look the same, so considering https://github.com/scrapinghub/spidermon/pull/150#issuecomment-484024765, what if we:
1. Remove `SPIDERMON_MIN_ITEMS` and only rely on `expected_item_scraped_count`
2. Set the default threshold to expect all expected items, meaning we don't raise not configured
So it's only the matter of writing `expected_item_scraped_count`

Of even better, let's forget about `expected_item_scraped_count` and use only thresholds to check if we scraped items based on`item_scraped_count` * threshold

Reply:
That makes.
Just the last case, I think I didn't get it. Mostly because against what am I going to compare `scraped_count * threshold`?

Reply:
Also, removing the settings breaks backward compatibility...

Reply:
Why do we need backwards compatibility?
I confused myself since we need the amount of scraped items and the amount of expected items as well.

Reply:
Because if there is someone using the library we cannot starting throwing errors when they update.
At first, we need to notify them that the API will change in the following release or so

Reply:
So let's notify them then :) Anyway, let's figure out if we want to keep it separate, make it a basic monitor first.

Reply:
We need to keep things  backward compatible as much as possible to avoid projects break after version updates. We have no idea who is using the library, so it is impossible to notify all of them. A good library will not break unless it is totally necessary, and this is not a change that would improve the library to justify it.

Reply:
Maybe it's not the change to break backwards compatibility.
All good libraries I am familiar with have deprecations so they still can change fast and have flexibility. Numpy, pandas, fastai to name a few.